File: hellozo0/hash/완주하지_못한_선수.java
package hellozo0.hash;

import java.util.HashMap;

public class 완주하지_못한_선수 {
    public String solution(String[] participant, String[] completion) {

        HashMap<String, Integer> map = new HashMap<>();
        String answer = "";

        for(int i = 0; i < participant.length; i++){
            if(map.containsKey(participant[i])){
                map.replace(participant[i], map.get(participant[i])+1);
            } else {
                map.put(participant[i], 1);
            }
        }

        for(int i = 0; i < completion.length; i++){
            if( map.get(completion[i]) != 1){
                map.replace(completion[i], map.get(completion[i])-1);
            } else {
                map.remove(completion[i]);
            }
        }

        for(String key : map.keySet() ){
            answer = key;
        }

        return answer;
    }
}

File: programmers/unanchoi/hash/완주하지못한선수.py
def solution(participant, completion):
    answer = ''
    participant.sort()
    completion.sort()

    for idx in range(len(completion)):
        if participant[idx] != completion[idx]:
            return participant[idx]
    return participant[-1]

print(solution(["leo", "kiki", "eden"], ["eden", "kiki"]))
